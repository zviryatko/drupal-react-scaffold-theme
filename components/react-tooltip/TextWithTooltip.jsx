import { Tooltip } from "react-tippy";

export const TextWithTooltip = ({ text, content }) => (
  <Tooltip
    html={<span className="react-tooltip__text">{text}</span>}
    className="react-tooltip__container"
    position="right"
    delay={100}
  >
    {content}
  </Tooltip>
);
