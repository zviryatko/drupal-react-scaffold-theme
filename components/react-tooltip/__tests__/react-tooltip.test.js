import { TextWithTooltip } from "../TextWithTooltip";
import { render } from "@testing-library/react";

describe("Component <TextWithTooltip />: ", () => {
  it("should match snapshot", () => {
    const { container, unmount } = render(
      <TextWithTooltip text='Test' content='Test content' />
    );

    expect(container.firstChild).toMatchSnapshot();

    unmount();
  });
});
