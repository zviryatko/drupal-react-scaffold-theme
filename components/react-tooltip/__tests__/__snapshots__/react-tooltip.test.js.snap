// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TextWithTooltip />:  should match snapshot 1`] = `
<div
  aria-describedby="tippy-tooltip-1"
  class="react-tooltip__container"
  data-tooltipped=""
  style="display: inline;"
>
  Test content
</div>
`;

exports[`Component <TextWithTooltip />:  should match snapshot 2`] = `
<div
  aria-describedby="tippy-tooltip-1"
  class="react-tooltip__container"
  data-tooltipped=""
  style="display: inline;"
>
  Test content
</div>
`;
